src/MB_Toolbox.php

```diff
@@ -238,31 +237,42 @@ private function curlPOSTauth($curlUrl, $post) {
    *  The values to POST.
    *
    * @return object $result
-   *   The results retruned from the cURL call.
+   *   The results returned from the cURL call.
    */
-  private function curlPOST($curlUrl, $post) {
+  public function curlPOST($curlUrl, $post) {
 
     $ch = curl_init();
     curl_setopt($ch, CURLOPT_URL, $curlUrl);
     curl_setopt($ch, CURLOPT_POST, count($post));
     curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($post));
     curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-    curl_setopt($ch, CURLOPT_HTTPHEADER,
-      array(
-        'Content-type: application/json',
-        'Accept: application/json'
-      )
-    );
     curl_setopt($ch,CURLOPT_CONNECTTIMEOUT, 3);
     curl_setopt($ch,CURLOPT_TIMEOUT, 20);
 
-    if (isset($this->auth->token)) {
-      curl_setopt($ch, CURLOPT_HTTPHEADER, array('Authorization', 'OAuth ' . $this->auth->token));
+    // Only add token and cookie values to header when values are available and
+    // the curlPOSTauth() method is making the POST request.
+    if (isset($this->auth->token) && debug_backtrace(2)[1]['function'] == 'curlPOSTauth') {
```

Comment:
`debug_backtrace(2)[1]['function'] == 'curlPOSTauth'` is not elegant solution at all? Why do you need an exception for it? Maybe it's better to add third argument to `curlPOST` instead?